python/ambassador/envoy/v2/v2route.py

```diff
@@ -92,7 +92,19 @@ def __init__(self, config: 'V2Config', group: IRHTTPMappingGroup, mapping: IRBas
         if envoy_route == 'prefix':
             match['prefix'] = route_prefix
         else:
-            match.update(regex_matcher(config, route_prefix))
+            # Cheat.
+            if config.ir.edge_stack_allowed and (self.get('_precedence', 0) == -1000000):
+                # Force the safe_regex engine.
+                match.update({
+                    "safe_regex": {
+                        "google_re2": {
+                            "max_program_size": 200,
+                        },
+                        "regex": route_prefix
+                    }
+                })
```

Comment:
What's the reason to cheat here?  The regex is simple enough that the unsafe portions of ecmascript regex shouldn't come in to play, if that's the regex engine that the ambassador module sets it to use.

Reply:
The cheating here isn't out of concern around safety: it's because the dictionary keys for safe-regex prefixes are different than they are for unsafe-regex prefixes, and it's easier to force it on setting than to look up both keys later.

(A more right answer would be to refactor things to remember the fallback route _a priori_, and carry that all the way through to the `V2Route`. I don't want to add more private members to `V2Route` though, and I don't want to refactor that class for more elegance right now.)